crates/node/src/service.rs

```diff
@@ -527,6 +533,14 @@ pub fn new_full(
         );
     }
 
+    if let Some(ref l1_messages_config) = l1_messages {
+        let config = L1MessagesWorkerConfig::try_from(l1_messages_config)?;
+        task_manager.spawn_essential_handle().spawn(
+            "L1 Messages",
+            Some(MADARA_TASK_GROUP),
+            mc_l1_messages::worker::run_worker(config, client, transaction_pool, db_backend),
```

Comment:
I think there is a small misunderstood there.
`task_manager.spawn_essential_handle().spawn` will spawn a task, once. The task will only be spawn once. When the function you are calling, here `mc_l1_messages::worker::run_worker` returns, the thread will die.

The design we want here is that this function you call should never return. It should contain an infinite loop, asynchronously pulling blocks from Ethereum. 

The way I read your impl of `run_worker` it will return after processing all the events since the last time it was called. But it's only called once at the node launch.

Instead you should call `ether.events()` inside an infinite loop

Reply:
@tdelabro this implementation provides an infinite loop. The way an event stream works is that it listens to specific events and does not end. So you do not need to worry here :)

Reply:
@tdelabro the only way this function can return earlier is potential issues with connection to L1 Contract or storage.

Reply:
But your infinite loop will end up yielding `None` forever once the first set of events is processed. It will never catch up to later l1 blocks

Reply:
I disagree, it catches later (future) blocks